### PR TITLE
KE2: Extract `null` literal

### PR DESCRIPTION
**Pull request by tamasvajk**

### CODE REVIEW

File: java/kotlin-extractor2/src/main/kotlin/entities/Expression.kt [hunk 1]
```diff
@@ -166,6 +166,33 @@ OLD: KE1
...
+    OLD: KE1
+            overrideId: Label<out DbExpr>? = null
+    */
+) =
+    // OLD: KE1: Was: exprIdOrFresh<DbNullliteral>(overrideId).also {
+    tw.getFreshIdLabel<DbNullliteral>().also {
+        val type = useType(t)
+        // Match Java by using a special <nulltype> for nulls, rather than Kotlin's view of this which is
+        // kotlin.Nothing?, the type that can only contain null.
+        val nullTypeName = "<nulltype>"
```
Comment (igfoo) [inline java/kotlin-extractor2/src/main/kotlin/entities/Expression.kt:186]: This isn't entirely consistent with Java, e.g.
```
public class Foo {
    void foo() {
        java.lang.Void v = true ? null : null;
    }
}
```
gives
```
| Foo.java:3:28:3:45 | ...?...:... | file://:0:0:0:0 | <nulltype> |
| Foo.java:3:35:3:38 | null | file://:0:0:0:0 | <nulltype> |
| Foo.java:3:42:3:45 | null | file://:0:0:0:0 | <nulltype> |
```
(I'm not sure we have a Java test for that; might be worth adding one)

I think what we want is to check for `Nothing?` in `useType`, and return `nullType` for the Java type there.

Comment (smowton) [reply java/kotlin-extractor2/src/main/kotlin/entities/Expression.kt:186]: To check: I was recently requested to patch KE1 per https://github.com/github/codeql/pull/17890 to get the type of null consistent with Java

Comment (igfoo) [reply java/kotlin-extractor2/src/main/kotlin/entities/Expression.kt:186]: Hmm, if Kotlin sees a function defined in Java as `public java.lang.Void foo()`, then does that look like a `Nothing?` return type in Kotlin?

Comment (igfoo) [reply java/kotlin-extractor2/src/main/kotlin/entities/Expression.kt:186]: KE2/KE1 not consistent with the java extractor

**Comment by tamasvajk** [comment]
I rebased this PR to fix a merge conflict.

File: java/kotlin-extractor2/src/main/kotlin/entities/Expression.kt (hunk 1 above)
Comment (tamasvajk) [reply java/kotlin-extractor2/src/main/kotlin/entities/Expression.kt]: If it's independent of KE2, then I think we should handle that inconsistency in a separate PR.

**Review by igfoo — APPROVED** [review]

**Pull request merged**